src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioDiagnosticAnalyzerService.cs

```diff
@@ -74,5 +128,155 @@ public VisualStudioDiagnosticAnalyzerService(VisualStudioWorkspace workspace, ID
             // unfortunately, we had to do this since ruleset editor and us are set to use this signature
             return map.ToDictionary(kv => kv.Key, kv => (IEnumerable<DiagnosticDescriptor>)kv.Value);
         }
+
+        private void OnRunCodeAnalysisForSelectedProjectStatus(object sender, EventArgs e)
+        {
+            var command = (OleMenuCommand)sender;
+
+            // We hook up the "Run Code Analysis" menu commands for CPS based managed projects.
+            // These commands are already hooked up for csproj based projects in StanCore, but those will eventually go away.
+            var visible = TryGetSelectedProjectHierarchy(out var hierarchy) &&
+                hierarchy.IsCapabilityMatch("CPS") &&
+                hierarchy.IsCapabilityMatch(".NET");
+            var enabled = false;
+
+            if (visible)
+            {
+                if (command.CommandID.ID == RunCodeAnalysisForSelectedProjectCommandId &&
+                    hierarchy.TryGetProject(out var project))
+                {
+                    // Change to show the name of the project as part of the menu item display text.
+                    command.Text = string.Format(ServicesVSResources.Run_Code_Analysis_on_0, project.Name);
+                }
+
+                enabled = !IsBuildActive();
+            }
+
+            if (command.Visible != visible)
+            {
+                command.Visible = visible;
+            }
+            if (command.Enabled != enabled)
+            {
+                command.Enabled = enabled;
+            }
+        }
+
+        private void OnRunCodeAnalysisForSelectedProject(object sender, EventArgs args)
+        {
+            if (TryGetSelectedProjectHierarchy(out var hierarchy))
+            {
+                RunAnalyzers(hierarchy);
+            }
+        }
+
+        public void RunAnalyzers(IVsHierarchy? hierarchy)
+        {
+            var project = GetProject(hierarchy);
+            var projectOrSolutionName = project?.Name ?? PathUtilities.GetFileName(_workspace.CurrentSolution.FilePath) ?? "Solution";
+
+            // Add a message to VS status bar that we are running code analysis.
+            var waitDialogMessage = string.Format(ServicesVSResources.Running_code_analysis_for_0, projectOrSolutionName);
+            var statusBar = _serviceProvider?.GetService(typeof(SVsStatusbar)) as IVsStatusbar;
+            statusBar?.SetText(waitDialogMessage);
+
+            // Force complete analyzer execution in background.
+            Task.Run(async () =>
```

Comment:
❕  This work needs to be tracked (IAsyncOperationListener)

Reply:
Done